air/src/passes/translate_from_mir.rs

```diff
@@ -604,6 +570,79 @@ impl AirBuilder<'_> {
     fn insert_op(&mut self, op: Operation) -> NodeIndex {
         self.air.constraint_graph_mut().insert_node(op)
     }
+
+    /// Extracts the trace access information from a given [Mir] `Boundary``.
+    fn extract_trace_from_boundary(
+        &self,
+        boundary: MirBoundary,
+    ) -> Result<MirTraceAccess, CompileError> {
+        let expected_trace_access_expr = boundary.expr.clone();
+        let Op::Value(value) = expected_trace_access_expr.borrow().deref().clone() else {
+            unreachable!(); // Raise diag
+        };
+
+        let trace_access = match value.value.clone() {
+            SpannedMirValue {
+                value: MirValue::TraceAccess(trace_access),
+                ..
+            } => trace_access,
+            SpannedMirValue {
+                value: MirValue::TraceAccessBinding(trace_access_binding),
+                span,
+            } => {
+                if trace_access_binding.size != 1 {
+                    self.diagnostics.diagnostic(Severity::Error)
+                                .with_message("invalid boundary constraint")
+                                .with_primary_label(span, "this has a trace access binding with a size greater than 1")
+                                .with_note("Boundary constraints require both sides of the constraint to be single columns.")
+                                .emit();
+                    return Err(CompileError::Failed);
+                }
+                MirTraceAccess {
+                    segment: trace_access_binding.segment,
+                    column: trace_access_binding.offset,
+                    row_offset: 0,
+                }
+            },
+            SpannedMirValue {
+                value: MirValue::BusAccess(bus_access), ..
+            } => {
+                let bus = bus_access.bus;
+                let name = bus.borrow().deref().name();
+                let column = self.bus_bindings_map.get(&name).unwrap();
+                MirTraceAccess::new(TraceSegmentId::Aux, *column, bus_access.row_offset)
+            },
+            _ => unreachable!("Expected TraceAccess or BusAccess, received {:?}", value.value), /* Raise diag */
+        };
+
+        Ok(trace_access)
+    }
+
+    /// Marks a boundary as constrained by the given trace access information.
+    fn mark_constrained_boundary(
+        &mut self,
+        trace_access: MirTraceAccess,
+        boundary: &MirBoundary,
+    ) -> Result<(), CompileError> {
```

Comment:
Could we expand the comment here as well? My understanding is that the reason we do this is to make sure we don't have duplicate boundary constraints. Or are there other reasons?